### PR TITLE
PLT-720: Added support for private access cidrs

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

File: spectrocloud/resource_cluster_eks.go [hunk 1]
```diff
@@ -162,10 +162,21 @@ func resourceClusterEks() *schema.Resource {
...
 						"public_access_cidrs": {
-							Type:     schema.TypeSet,
-							Optional: true,
-							ForceNew: true,
-							Set:      schema.HashString,
+							Type:        schema.TypeSet,
+							Optional:    true,
+							ForceNew:    true,
+							Set:         schema.HashString,
+							Description: "List of CIDR blocks that define the allowed public access to the resource. Requests originating from addresses within these CIDR blocks will be permitted to access the resource. All other addresses will be denied access.",
```
Comment (nikchern) [inline spectrocloud/resource_cluster_eks.go:169]: what is the default value @SivaanandM ?

File: spectrocloud/resource_cluster_eks.go [hunk 2]
```diff
@@ -162,10 +162,21 @@ func resourceClusterEks() *schema.Resource {
...
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"private_access_cidrs": {
+							Type:        schema.TypeSet,
+							Optional:    true,
+							ForceNew:    true,
+							Set:         schema.HashString,
+							Description: "List of CIDR blocks that define the allowed private access to the resource. Only requests originating from addresses within these CIDR blocks will be permitted to access the resource.",
```
Comment (nikchern) [inline spectrocloud/resource_cluster_eks.go:179]: same here - we have to establish a default value.

File: spectrocloud/resource_cluster_eks_flatten_test.go [hunk 3]
```diff
@@ -178,3 +179,65 @@ func TestFlattenClusterConfigsEKS(t *testing.T) {
 		})
 	}
 }
+
+func TestFlattenClusterConfigsEKSPrivateCIDRS(t *testing.T) {
```
Comment (nikchern) [inline spectrocloud/resource_cluster_eks_flatten_test.go:183]: we also need a coverage for toEksCluster() part @SivaanandM

**Review by nikchern — CHANGES_REQUESTED** [review]

File: spectrocloud/resource_cluster_eks.go (hunk 1 above)
Comment (SivaanandM) [reply spectrocloud/resource_cluster_eks.go:169]: we are not using this attribute for cluster it used in resource_workspace

Comment (SivaanandM) [reply spectrocloud/resource_cluster_eks.go:169]: if we set default it always show as a change in day 2 operation

Comment (SivaanandM) [reply spectrocloud/resource_cluster_eks.go:169]: we don’t have any default values for private_access_cidrs  and public_access_cidrs, its a list option

File: spectrocloud/resource_cluster_eks_flatten_test.go (hunk 3 above)
Comment (SivaanandM) [reply spectrocloud/resource_cluster_eks_flatten_test.go:183]: we don’t have any default values for private_access_cidrs  and public_access_cidrs, its a list option

File: spectrocloud/resource_cluster_eks.go (hunk 2 above)
Comment (SivaanandM) [reply spectrocloud/resource_cluster_eks.go:179]: we don’t have any default values for private_access_cidrs  and public_access_cidrs, its a list option

**Comment by SivaanandM** [comment]
@nikchern we don't have any hash generated for cloud_config attributes

File: spectrocloud/resource_cluster_eks_flatten_test.go (hunk 3 above)
Comment (SivaanandM) [reply spectrocloud/resource_cluster_eks_flatten_test.go:183]: I have covered the unit test for private cidr change and will raise a separate ticket for toEksCluster() unit test PLT-763

**Review by nikchern — APPROVED** [review]

**Pull request merged**